projects/cm_mcu/LocalTasks.c

```diff
@@ -866,16 +883,23 @@ void snapdump(struct dev_i2c_addr_t *add, uint8_t page, uint8_t snapshot[32], bo
       log_error(LOG_SERVICE, "error reset\r\n");
     }
   }
-  xSemaphoreGive(i2c1_sem);
+
+  // if we have a semaphore, give it
+  if (xSemaphoreGetMutexHolder(i2c1_sem) == xTaskGetCurrentTaskHandle()) {
+    xSemaphoreGive(i2c1_sem);
+  }
 }
 
 // Initialization function for the LGA80D. These settings
 // need to be called when the supply output is OFF
 // this is currently not ensured in this code.
 void LGA80D_init(void)
 {
-  while (xSemaphoreTake(i2c1_sem, (TickType_t)10) == pdFALSE)
-    ;
+  // grab the semaphore to ensure unique access to I2C controller
+  if (acquireI2CSemaphore(i2c1_sem) == pdFAIL) {
+    log_warn(LOG_SERVICE, "could not get semaphore in time\r\n");
```

Comment:
I think this is one we _cannot_ skip and should use the blocking call. If the LGA80D are not initialized properly we will not be able to use the FPGAs.

Reply:
got it. I will change continue to block for this one and all other Init functions.